### ISSUE TITLE
Bug about view table when use version 7.0.3.1

### ISSUE THREAD

**Issue by behemoth-truongnd** [rails/rails#45666]
Hi everyone!
in the database i create a view table
```
CREATE VIEW view_favorites AS
        SELECT account_id, targetable_id, "" as youtube_id, targetable_type, is_favorited, created_at, updated_at, id as favorite_id, "" as favorite_youtube_id FROM favorites
        UNION ALL
        SELECT account_id, "" , youtube_id, "Youtube", is_favorited, created_at, updated_at, "", id FROM favorite_youtubes;
```

I am getting error when using views table

```
NoMethodError: undefined method `match' for nil:NilClass

          match = create_table_info(table_name).match(/`#{field_name}` (.+) DEFAULT ('|\d+|[A-z]+)/)
                                               ^^^^^^
/app/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.3.1/lib/active_record/connection_adapters/mysql/schema_statements.rb:162:in `default_type'
```

### System configuration
**Rails version**:
7.0.3.1
**Ruby version**:
3.1.1

**Comment by yahonda** [comment]
I think this is a duplicate with #45062 and fixed in the 7-0-stable branch via #45067, not released as 7.0.z yet.

**Comment by behemoth-truongnd** [comment]
@yahonda 
oh that right.
thank you so much. I will close this issue
